### PR TITLE
feat: user portfolio list page

### LINKED ISSUE #15: [UI] Individual portfolio default page
## Features

- this is what the user sees when they click on a specific portfolio
CleanShot 2023-10-16 at 21 32 41@2x

### Core

- default empty page with no transactions/asset allocation
- add transaction button
- edit button

## Potential Issues

- might need to abstract some components to reuse from the dashboard page

### PR DESCRIPTION
**Pull request by jeromeandrewong**
closes #15 
## Demo
CleanShot 2023-11-03 at 03 40 01

## Implementation
- `/api/portfolio/[id]` is the route to get all portfolios for a user given user's id
  - this route calls our backend endpoint at `localhost:8080/api/users/<id>/portfolios`
- each portfolio routes to its individual portfolio page at `/portfolios/[name].tsx`, to be implemented by #17 #16

## Todo
- replace hardcoded user id after unblocked by #28 #14 

## Testing
- add `NEXT_PUBLIC_API_URL=http://localhost:8080` into `.env`
- replace line 71-78 on `init-sql` with this
```
INSERT INTO prod.user (email, username) VALUES ('test@test.com', 'user1');
INSERT INTO prod.user (email, username) VALUES ('user2@example.com', 'user2');
INSERT INTO prod.user (email, username) VALUES ('user3@example.com', 'user3');


INSERT INTO prod.portfolio (user_id, portfolio_name, description) VALUES (1, 'Flagship Portfolio', 'Multi-manager, multi-strategy equity fund focused on Asia.');
INSERT INTO prod.portfolio (user_id, portfolio_name, description) VALUES (1, 'China A-Shares Plus Fund', 'Quantitative long-only strategies in Chinese A-Shares');
INSERT INTO prod.portfolio (user_id, portfolio_name, description) VALUES (1, 'Balanced Alternative Assets Fund', 'Designed to deliver protection for equity portfolios and generate high positive returns during market downtrends');
INSERT INTO prod.portfolio (user_id, portfolio_name, description) VALUES (2, 'ESG Portfolio', 'This is my ESG portfolio entry.');
INSERT INTO prod.portfolio (user_id, portfolio_name, description) VALUES (2, 'SGX Portfolio', 'This is a special portfolio entry in Singapore!');
```

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,3 @@
 NEXT_PUBLIC_SUPABASE_ANON_KEY=SUPABASE_ANON_KEY
 NEXT_PUBLIC_SUPABASE_URL=https://PROJECT_ID.supabase.co
+NEXT_PUBLIC_API_URL=http://localhost:8080
```
Comment (DavidLHW) [inline .env.example:3]: server url shouldn't be exposed with `NEXT_PUBLIC_` since FE side don't need context to this endpoint

**Review by DavidLHW — APPROVED** [review]
minor comment on the `.env`, other than that, CLGTM @ 6656a1c43e1584286ce2186b2b405742e04725ec!

File: .env.example (hunk 1 above)
Comment (jeromeandrewong) [reply .env.example]: Thanks for catching this, fixed [here](https://github.com/is442oop/portfolio-analyzer-web-ui/pull/29/commits/2fca2d79e8c9271df9e8263a37ce7c297365f776)

**Review by DavidLHW — APPROVED** [review]
CLGTM @ 2fca2d79e8c9271df9e8263a37ce7c297365f776!

**Pull request merged**